src/fhirtypes/common.ts

```diff
@@ -218,6 +218,28 @@ export function setPropertyOnInstance(
         } else {
           if (typeof assignedValue === 'object') {
             Object.assign(current[key][index], assignedValue);
+            // check if we have used modifier extensions correctly
+            if ((key === 'extension' || key === 'modifierExtension') && assignedValue.url != null) {
```

Comment:
One thing I'm noticing about the way this works is that it only works when we are assigning the entire `Extension` as an `Instance`. Consider the example below:
```
Extension: Foo
* value[x] only boolean
* . ?!

Instance: FooInstance
InstanceOf: Foo
Usage: #inline
* valueBoolean = true

Instance: 😄
InstanceOf: Patient
* extension[Foo].valueBoolean = true

Instance: 😔
InstanceOf: Patient
* extension[0] = FooInstance
```
In this example, the 😄  and 😔  Instances generate the same JSON, but for the 😔  Instance we log an error, while for the 😄  we don't. My guess would be that the 😄  would actually be the more common way to accomplish this. Do you think we should also log an error in that case?

Reply:
Oof... yeah, I can't really argue against this. This is a little more intense, but I'll see what I can conjure up.

Reply:
Also, writing the modifier rule as `* . ?!` is a lot more concise than my `* . ^isModifier = true`, so thank you for that as well.